ament_tools/verbs/list_dependencies.py

```diff
@@ -0,0 +1,88 @@
+# Copyright 2017 Open Source Robotics Foundation, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+
+from ament_tools.helper import argparse_existing_dir
+from ament_tools.packages import find_packages
+
+
+def prepare_arguments(parser):
+    parser.add_argument(
+        'basepath',
+        nargs='?',
+        type=argparse_existing_dir,
+        default=os.curdir,
+        help='Base paths to recursively crawl for packages',
+    )
```

Comment:
Having an optional positional argument changes the semantic of the first passed argument based on if a second argument is passed or not. I think that is counter intuitive and I would suggest using a non-positional argument for the base path instead.